README.md

```diff
@@ -1 +1,394 @@
+
+Create a JSON file that contains all the information needed to connect and authenticate to Azure:
+
+```console
+# Create service principal with Owner role
+az ad sp create-for-rbac --sdk-auth --role Owner --name platform-ref-azure > crossplane-azure-provider-key.json
+```
+
+Take note of the `clientID` value from the JSON file that we just created, and save it to an environment variable:
+
+```console
+export AZURE_CLIENT_ID=<clientId value from json file>
+```
+
+Now add and grant the required permissions to the service principal that will allow it to manage the necessary resources in Azure:
+
+```console
+# add required Azure Active Directory permissions
+az ad app permission add --id ${AZURE_CLIENT_ID} --api 00000002-0000-0000-c000-000000000000 --api-permissions 1cda74f2-2616-4834-b122-5cb1b07f8a59=Role 78c8a3c8-a07e-4b9e-af1b-b5ccab50a175=Role
+
+# grant (activate) the permissions
+az ad app permission grant --id ${AZURE_CLIENT_ID} --api 00000002-0000-0000-c000-000000000000 --expires never
+```
+
+You might see an error similar to the following, but that is OK, the permissions should have gone through still:
+
+_Operation failed with status: 'Conflict'. Details: 409 Client Error: Conflict for url: https://graph.windows.net/e7985bc4-a3b3-4f37-b9d2-fa256023b1ae/oauth2PermissionGrants?api-version=1.6_
+
+Finally, you need to grant admin permissions on the Azure Active Directory to the service principal because it will need to create other service principals for your AKSCluster:
+
+```console
+# grant admin consent to the service princinpal you created
+az ad app permission admin-consent --id "${AZURE_CLIENT_ID}"
+```
+
+_Note: You might need Global Administrator role to Grant admin consent for Default Directory. Please contact the administrator of your Azure subscription. To check your role, go to Azure Active Directory -> Roles and administrators. You can find your role(s) by clicking on Your Role (Preview)_
+
+After these steps are completed, you should have the following file on your local filesystem:
+
+- crossplane-azure-provider-key.json
+
+#### Setup Azure ProviderConfig
+
+Before creating any resources, we need to create and configure an Azure cloud provider resource in Crossplane, which stores the cloud account information in it. All the requests from Crossplane to Azure Cloud will use the credentials attached to this provider resource. The following command assumes that you have a crossplane-azure-provider-key.json file that belongs to the account you’d like Crossplane to use.
+
+Now we’ll create our Secret that contains the credential and ProviderConfig resource that refers to that secret:
+
+```console
+kubectl create secret generic azure-account-creds -n upbound-system --from-file=credentials=./crossplane-azure-provider-key.json
+kubectl -n upbound-system apply -f examples/azure-default-provider.yaml
```

Comment:
This line gives the following error:
```
error: error validating "examples/azure-default-provider.yaml": error validating data: ValidationError(ProviderConfig.spec.credentials.secretRef): missing required field "namespace" in io.crossplane.azure.v1beta1.ProviderConfig.spec.credentials.secretRef; if you choose to ignore these errors, turn validation off with --validate=false
```
I believe we should add the namespace field back into the `azure-default-provider.yaml` file.

Reply:
Following lines are also creating the provider. I suggest to remove the line above and update `provider.yaml` filename to `examples/azure-default-provider.yaml` in below.

Reply:
thanks for the spotting. there was a leftover provider apply from a block I deleted. Got that and replaced namespace reference for provider secret ref. I have the comment from Jared open still. I am not understanding something so need to get clarification.